app/models/podcast_import.rb

```diff
@@ -13,89 +13,160 @@ class PodcastImport < BaseModel
+    episode_imports.all? { |e| e.status == EpisodeImport::COMPLETE }
+  end
+
+  def some_failed?
+    episode_imports.any? { |e| e.status == EpisodeImport::FAILED }
+  end
+
   def retry!
-    update_attributes(status: 'retrying')
+    update_attributes(status: RETRYING)
     import_later
   end
 
-  def import_later
-    PodcastImportJob.perform_later self
+  def import_later(import_series = true)
+    PodcastImportJob.perform_later(self, import_series)
   end
 
-  def import
-    update_attributes!(status: 'started')
+  def import_series!
+    update_attributes!(status: STARTED)
 
     # Request the RSS feed
     get_feed
-    update_attributes!(status: 'feed retrieved')
+    update_attributes!(status: FEED_RETRIEVED)
 
     # Create the series
     create_or_update_series!
-    update_attributes!(status: 'series created')
+    update_attributes!(status: SERIES_CREATED)
+  rescue StandardError => err
+    update_attributes(status: FAILED)
+    raise err
+  end
+
+  def import_episodes!
+    # Request the RSS feed again
+    get_feed
 
     # Update podcast attributes
     create_or_update_podcast!
-    update_attributes!(status: 'podcast created')
+    update_attributes!(status: PODCAST_CREATED)
 
     # Create the episodes
-    update_attributes!(status: 'importing')
+    update_attributes!(status: IMPORTING)
     create_or_update_episode_imports!
   rescue StandardError => err
-    update_attributes(status: 'failed')
+    update_attributes(status: FAILED)
     raise err
   end
 
-  def entry_audio_files(entry)
-    if config[:audio] && config[:audio][entry[:entry_id]]
-      { files: (config[:audio][entry[:entry_id]] || []) }
-    elsif enclosure = enclosure_url(entry)
-      { files: [enclosure] }
-    end
+  def import
+    import_series!
+    import_episodes!
   end
 
   def create_or_update_episode_imports!
-    feed.entries.map do |entry|
-      entry_hash = feed_entry_to_hash(entry)
-      audio_files = entry_audio_files(entry_hash)
-      get_or_create_template(audio_files, entry_hash['enclosure'].type)
-      episode_import = create_or_update_episode_import!(entry_hash, audio_files)
-      episode_import.import_later
+    feed_entries, entries_with_dupe_guids = parse_feed_entries_for_dupe_guids
+
+    update_attributes(feed_episode_count: feed_entries.length)
+
+    episode_imports.having_duplicate_guids.destroy_all
+
+    created_imports = feed_entries.map do |entry|
+      episode_import = create_or_update_episode_import!(entry)
+    end
+
+    enqueue_episode_import_jobs(created_imports)
+
+    created_imports += entries_with_dupe_guids.map do |entry|
+      create_or_update_episode_import!(entry, has_duplicate_guid = true)
+    end
+
+    created_imports
+  end
+
+  def enqueue_episode_import_jobs(created_imports)
+    messages = created_imports.map do |ei|
+      job = EpisodeImportJob.new(ei)
+      msg = {}
+      msg[:message_body] = job.serialize
+      msg[:message_attributes] = {
```

Comment:
So this is manually queuing messages for shoryuken?  There's no built in way to schedule those?

Reply:
So, yeah. I'm not super happy about this, but this change shaved about 1-2 minutes off the podcast import run (which now runs in about 20 seconds for n=1000 episodes).

The story is that enqueuing jobs on SQS is _really_ slow. There is a bulk api for shoryuken workers I think.  Maybe I should have ported the podcast import job to become a shoryuken worker instead?

Reply:
That is quite a bit quicker!  I guess my main objection is knowing the arbitrary structure of the messages here.

I guess I'm fine with leaving this in, as long as it's marked as TODO/MAGIC or something.  And maybe a ticket to come back and do a refactor at some point.